arenarobot/service/processor/apriltag_detector.py

```diff
@@ -0,0 +1,178 @@
+"""
+apriltag_detector.py: reads frames, detects apriltags, and returns pose.
+
+Created by John Pedraza on 6/7/22.
+
+Copyright (c) 2022, The CONIX Research Center
+All rights reserved.
+
+This source code is licensed under the BSD-3-Clause license found in the
+LICENSE file in the root directory of this source tree.
+"""
+
+from json import JSONEncoder, dumps, loads
+import numpy as np
+import cv2
+from paho.mqtt.client import MQTTMessage
+from transformations import euler_matrix, quaternion_matrix
+from dt_apriltags import Detector
+from arenarobot.service.processor import ArenaRobotServiceProcessor
+from scipy.spatial.transform import Rotation as R
+
+# pylint: disable=too-many-instance-attributes
+class ArenaRobotServiceProcessorApriltagDetector(ArenaRobotServiceProcessor):
+    """Apriltag Detector processor class for the ARENA."""
+
+    DEVICE_INSTANCE_PROCESSOR_TYPE = "apriltag_detector"
+
+    # pylint: disable=too-many-arguments
+    # pylint: disable=too-many-statements
+    def __init__(self, processor_apriltag_detector_topic: str,
+                 processor_apriltag_detector_instance_name: str = None,
```

Comment:
These two parameters would be for receiving data from another topic (which is not in use here), so these can be removed.